### ISSUE TITLE
Bluebird 3 Promise warning

### ISSUE THREAD

**Issue by luthv** [NodeRedis/node_redis#993]
Using the latest Bluebird 3.3.3 with this recommended way of promisifying redis client:
```javascript
var redis = require('redis');
Promise.promisifyAll(redis.RedisClient.prototype);

function getList(key){
   var client = redis_connection();
   return client.lrangeAsync(key, 0, -1);
}
```
Calling the above getList function triggers this warning by Bluebird:
_Warning: a promise was created in a handler but was not returned from it_

I'm new to using redis and promise so it could be that I'm using it the wrong way, so what is the correct way of turning the redis command to a promisified one?

**Comment by BridgeAR** [comment]
I guess the redis_connection fires promises too, so the error is going to be printed. Could you please show the redis_connection code?

**Comment by luthv** [comment]
basically it just create a new redis client
```javascript
function new_connection () {
  var redis_con;

  try {
    redis_con = redis.createClient(rc.port, rc.host);
    if(rc.auth){
      redis_con.auth(rc.auth);
    }
  } catch (e) {
    console.log(e);
  } finally {

  }
  return redis_con;
}
```
If it makes a difference, I'm testing it on Windows 10, node 5.3.0, and the redis client is connected to a rediscloud account.

**Comment by luthv** [comment]
here is the partial stack trace included in the warning
```
Warning: a promise was created in a handler but was not returned from it                                       
    at bound (domain.js:287:14)                                                                                
    at Command.runBound (domain.js:300:12)                                                                     
    at bound (domain.js:287:14)                                                                                
    at Command.runBound (domain.js:300:12)                                                                     
    at RedisClient.return_reply (\node_modules\redis\index.js:664:25)            
    at JavascriptReplyParser.reply_parser.send_reply (\node_modules\redis\index.j
s:332:14)                                                                                                      
    at JavascriptReplyParser.run (\node_modules\redis\lib\parsers\javascript.js:1
32:18)                                                                                                         
    at JavascriptReplyParser.execute (\node_modules\redis\lib\parsers\javascript.
js:107:10)                                                                                                     
    at Socket.<anonymous> (\node_modules\redis\index.js:131:27)                  
    at emitOne (events.js:77:13)                                                                               
    at Socket.emit (events.js:169:7)                                                                           
    at readableAddChunk (_stream_readable.js:146:16)                                                           
    at Socket.Readable.push (_stream_readable.js:110:10)                                                       
From previous event:                                                                                           
    at Object.getList (cache.js:98:19)  <-- this is my code, 
	
	line 98 is return client.lrangeAsync(key, 0, -1) from the above snippet
```

**Comment by BridgeAR** [comment]
I can't reproduce this with the code you provided. There has to be some other code you use that triggers that warning.

**Comment by luthv** [comment]
thanks for following this up, I've no luck in reproducing this behavior outside of my project either. So I guess it's like what you said, something else in my code that caused it. I'll update this issue once I found out more.

**Comment by luthv** [comment]
I'm feeling ashamed of myself right now, I finally found the problem, it was indeed a problem in my code that consumed the lrangeAsync promise. I didn't realize that I'm calling another promise in my other if branch statement, changing it to return the second promise instead of just calling it directly fixed the problem. 

What I failed to understand was the bluebird warning stack only reports up to the first promise which is from redis promise. I guess this is a problem with me being non native English speaker, a better more concise wording will probably help a lot here, something like:

`Warning: a **new** promise was created in a handler but was not returned from it`

Oh well, anyway thanks a lot for your patience and effort, I really appreciates it.

**Comment by BridgeAR** [comment]
I'm glad I could help you :)

Btw: you could provide auth directly to the createClient function instead of calling it on your own and you do not have to wrap it in a try catch block. Any documented way to pass the options won't ever throw. Only malformed options are going to throw and you'd realize that while implementing / testing it.
